<meta name="viewport" content="width=device-width,initial-scale=1"><nav class="navbar navbar-expand-lg navbar-inverse noMargin navbar-fixed-top"><div class="container"><div class="navbar-header"><button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar"><span class="sr-only">Toggle navigation</span> <span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span></button> <a class="navbar-brand" href="/"><img ng-src="/static/wiki/img/cb-logo-long.svg" style="height:35px"></a></div><div id="navbar" class="navbar-collapse collapse"><ul class="nav navbar-nav navbar-right"><li><a data-target="#about_modal" data-toggle="modal" class="btn">About</a></li><li><a data-target="#pub_modal" data-toggle="modal" class="btn">Publications</a></li><li><a data-target="#advisors_modal" data-toggle="modal" class="btn">The Team</a></li><li><a data-target="#faq_modal" data-toggle="modal" class="btn">FAQ</a></li><li><a data-target="#support_modal" data-toggle="modal" class="btn">Support Us</a></li><li class="dropdown bg-purple"><a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">More WikiGenomesDB <span class="caret"></span></a><ul class="dropdown-menu bg-pink" style="max-width:200px;min-height:150px"><li><a class="text-light" target="_blank" href="https://chlambase.org/">ChlamBase</a></li><li><a class="text-light" target="_blank" href="https://myxobase.org/">MyxoBase</a></li><li><a class="text-light" target="_blank" href="http://13.58.104.209/">FishBase</a></li><li><a class="text-light" target="_blank" href="http://wikigenomes.org/">WikiGenomesDB</a></li></ul></li></ul></div><form class="navbar-form navbar-right" ng-if="$ctrl.page == 'main'"><oauth-view></oauth-view></form><form class="navbar-form navbar-right" ng-if="$ctrl.page != 'landing'" style="color:#f5f5f5;margin-right:10px"><keyword-form></keyword-form></form></div></nav><div id="about_modal" class="modal fade" role="dialog"><div class="modal-dialog modal-lg"><div class="modal-content"><div class="modal-header bg-purple"><button type="button" class="close text-light" data-dismiss="modal">&times;</button><h4 class="modal-title text-pink-light">About ChlamBase</h4></div><div class="modal-text-color modal-body bg-dark text-light padding20"><div class="text-center"><img src="static/wiki/img/cb-logo.svg" style="width:150px"></div><p>ChlamBase.org, a fork of <a href="http://wikigenomes.org">WikiGenomes.org</a>, is an open source, user-curated database of functional annotations for 4 model <i>Chlamydia</i> species: <i>C. trachomatis 434/BU</i> ( LGV L2 ), <i>C. trachomatis D/UW-3/CX</i>, <i>C. muridarum Nigg</i>, and <i>C. pneumoniae CWL029</i>. A primary goal of this portal is to provide a powerful user-oriented experience tailored to the research interests of the <i>Chlamydia</i> research community, and to the unique biology of <i>Chlamydia</i>. A central feature of ChlamBase is to allow users to view, add, and edit evidence-based gene annotations, engineered mutant strains, orthologous gene comparisons, developmental gene expression, protein interactions, functions, processes, and more. Furthermore, ChlamBase allows visualization of the organism genome using JBrowse at the top of the main page, and also provides a widget for sequence alignment of either dna or amino acids of chlamydia orthologs.</p><div class="text-center"><h2 class="text-pink-light smallCaps">Funding</h2><p class="text-light">Support for ChlamBase comes NIH/NIGMS R01GM089820, PI Andrew Su.</p></div></div></div></div></div><div id="wikigenomes_modal" class="modal fade" role="dialog"><div class="modal-dialog modal-lg"><div class="modal-content"><div class="modal-header bg-purple"><button type="button" class="close text-light" data-dismiss="modal">&times;</button><h4 class="modal-title text-pink-light">WikiGenomes</h4></div><div class="modal-text-color modal-body bg-dark text-light padding20"><p>WikiGenomes is a portal to the biological content in the <a href="http://wikidata.org">Wikidata</a> project. It currently hosts the genes, proteins, and a variety of functional annotations for 120 bacteria genomes, as well as several species of eukaryote. WikiGenomes is designed to be both a resource for basic researchers to browse the collective knowledge of a variety of well studied organisms, and a technology layer that allows the user to add to that collective knowledge. To support this, WikiGenomes provides defined editing forms that prompt the user to provide key information and references. WikiGenomes then adds that knowledge to the Wikidata knowledge graph of genes, diseases, and drugs.</p></div></div></div></div><div id="pub_modal" class="modal fade" role="dialog"><div class="modal-dialog modal-lg"><div class="modal-content"><div class="modal-header bg-purple"><button type="button" class="close text-light" data-dismiss="modal">&times;</button><h4 class="modal-title text-pink-light">Publications</h4></div><div class="modal-text-color modal-body bg-dark text-light padding20"><a target="_blank" href="https://academic.oup.com/database/article-lookup/doi/10.1093/database/bax025">Tim E. Putman Sebastien Lelong, Sebastian Burgstaller-Muehlbacher, Andra Waagmeester, Colin Diesh, Nathan Dunn, Monica Munoz-Torres, Gregory S. Stupp, Chunlei Wu, Andrew I. Su, and Benjamin M. Good. 2017. <strong class="text-pink-light">â€œWikiGenomes: An Open Web Application for Community Consumption and Curation of Gene Annotation Data in Wikidata.â€� Database 2017 (March). doi:10.1093/database/baw028.</strong></a><h4>Related Projects</h4><a href="https://en.wikipedia.org/wiki/Portal:Gene_Wiki">Gene Wiki</a><br><a href="https://www.wikidata.org/wiki/Wikidata:WikiProject_Molecular_biology">Wikidata Project Molecular Biology</a><br><a href="https://www.wikidata.org/wiki/Wikidata:WikiProject_Microbiology">Wikidata Project Microbiology</a><br></div></div></div></div><div id="advisors_modal" class="modal fade" role="dialog"><div class="modal-dialog modal-lg"><div class="modal-content"><div class="modal-header bg-purple"><button type="button" class="close text-light" data-dismiss="modal">&times;</button><h4 class="modal-title text-pink-light">The Team</h4></div><div class="modal-text-color modal-body bg-dark text-light padding20"><div class="flexParent"><div class="flexChild text-center"><img width="40%" style="border-radius:100%" src="https://avatars0.githubusercontent.com/u/2635409?s=460&v=4"><h5 class="text-blue smallCaps">Andrew Su / Professsor</h5><p>Scripps Research</p><p><a href="https://twitter.com/andrewsu?ref_src=twsrc%5Etfw" class="twitter-follow-button" data-show-count="false">Follow @andrewsu</a></p></div><div class="flexChild">Andrew is a Professor at Scripps Research in the Department of Integrative Structure and Computational Biology. His research focuses on building and applying bioinformatics infrastructure for biomedical discovery. His research has a particular emphasis on leveraging crowdsourcing for genetics and genomics. Representative projects include the Gene Wiki, BioGPS, MyGene.Info, and Mark2Cure, each of which engages “the crowd” to help organize biomedical knowledge. These resources are collectively used millions of times every month by members of the research community, by students, and by the general public.</div></div><hr><div class="flexParent"><div class="flexChild text-center"><img width="40%" style="border-radius:100%" src="https://media.licdn.com/dms/image/C5103AQFZJM_yeVeuWQ/profile-displayphoto-shrink_800_800/0?e=1542844800&v=beta&t=TwBkUrRxZp6kbVUp8IH1CgRY9weSyeYPBenqiF4bQLk"><h5 class="text-blue smallCaps">Tim Putman / Research Associate</h5><p>Scripps Research</p></div><div class="flexChild">Tim Putman obtained his PhD in Molecular and Cellular Biology in microbial genomics from Oregon State University in June of 2015 and joined the Su Lab as a research associate in August of the same year. His project involved modeling and linking microbial data in Wikidata. He left the lab in July, 2017 and is now a Research Assistant Professor and Bioinformatics Software Developer at Oregon Health and Science University.</div></div><hr><div class="flexParent"><div class="flexChild text-center"><img width="40%" style="border-radius:100%" src="https://pbs.twimg.com/profile_images/702369628706762753/rkefNcjC_400x400.png"><h5 class="text-blue smallCaps">Kevin Hybiske, PhD / Assistant Professor</h5><p>University of Washington</p></div><div class="flexChild">Assistant Professor, Division of Allergy and Infectious Diseases, Department of Medicine; Adjunct Assistant Professor, Department of Microbiology University of Washington The Hybiske Lab investigates the interactions between intracellular pathogens and host cells with a specific focus on the molecular, cellular and genetic basis of infectious diseases.</div></div><hr><div class="flexParent"><div class="flexChild text-center"><img width="40%" style="border-radius:100%" src="http://sulab.org/wp-content/uploads/2018/07/derek_jow-300x243.jpg"><h5 class="text-blue smallCaps">Derek Jow / Student UCSD</h5><p>University of California, San Diego</p></div><div class="flexChild">Currently studying Computer Science with a specialization in Bioinformatics at University of California, San Diego.</div></div><hr><div class="flexParent"><div class="flexChild text-center"><img width="40%" style="border-radius:100%" src="https://avatars1.githubusercontent.com/u/23092057?s=460&v=4"><h5 class="text-blue smallCaps">Marco Cano / Research Programmer II</h5><p>Scripps Research</p></div><div class="flexChild">Web programmer currently working for the Su/Wu Lab at Scripps Research. Current projects include: http://BioThings.io and http://SmartAPI.info</div></div><div class="text-center"><h2 class="text-pink-light smallCaps">Advisors</h2><ul><li>Scott Hefty (University of Kansas, USA)</li><li>Daniel Rockey (Oregon State University, USA)</li></ul></div></div></div></div></div><div id="support_modal" class="modal fade" role="dialog"><div class="modal-dialog modal-lg"><div class="modal-content"><div class="modal-header bg-purple"><button type="button" class="close text-light" data-dismiss="modal">&times;</button><h4 class="modal-title text-pink-light">Support Us</h4></div><div class="modal-text-color modal-body bg-dark text-light padding20"><p>This site is a free service and we would like it to remain that way. If you found us useful (even just at one point), please help support the project by demonstrating its value to our funders by doing as many of the following as possible:</p><ul><li><script async defer="defer" src="https://buttons.github.io/buttons.js"></script>Follow us on GitHub <a class="github-button" href="https://github.com/SuLab/WikiGenomesBase" data-size="large" data-show-count="true" aria-label="Star SuLab/WikiGenomesBase on GitHub">Star</a></li><li>Use our branding to spread the word about our work.</li></ul></div></div></div></div><div id="faq_modal" class="modal fade" role="dialog"><div class="modal-dialog modal-lg"><div class="modal-content"><div class="modal-header bg-purple"><button type="button" class="close text-light" data-dismiss="modal">&times;</button><h4 class="modal-title text-pink-light">FAQ</h4></div><div class="modal-text-color modal-body bg-dark text-light padding20">1. <strong class="text-pink-light">I want to get a list of all chlamydia genes that have annotated cellular components.</strong><br>Use the "advanced search" option. Click the "advanced search" dropdown and select the box next to "cellular component" in the first column. Then click the search button.<br><br>2. <strong class="text-pink-light">I want to get a list of all chlamydia genes that have the cellular component "pathogen-containing vacuole membrane".</strong><br>Use the "advanced search" option. Click the "advanced search" dropdown and select the box next to "cellular component" in the first column. Then, type in "pathogen-containing vacuole membrane" next to the checkbox. A list of all matching cellular components will appear, and select the best match. Now, click the search button.<br><br>3. <strong class="text-pink-light">I want a list of chlamydia proteins that have molecular function "SH2 domain binding" AND localized to the "extracellular region".</strong><br>Use the "advanced search" option. Click the "advanced search" dropdown and select the box next to "molecular function" in the first column. Then, type in "sh2" next to the checkbox. "SH2 domain binding" will appear from the dropdown, and select the best match. Click the checkbox next to cellular component and type "extracellular". A dropdown will appear with "extracellular region". Select it, and now, click the search button.<br><br>4. <strong class="text-pink-light">How do I get to the advanced search page?</strong><br>There are several ways. If you are on the landing page, you can do a direct advanced search by using the search bar at the top of the page. If you are on any other page, there is a search bar in the upper right corner. You can type in a keyword you are looking for, or simply leave it blank and click the search button. It will redirect you to the advanced search page.<br><br>5. <strong class="text-pink-light">How do I get to the gene page for the chlamydia gene with locus tag CTL0260?</strong><br>The easiest way is to use the search bar. Start typing CTL0260 and a dropdown will appear. Click CTL0260 and you will be redirected to its page. The other way is to use the browsing page. From the landing page, click the "C. trachomatis 434/BU" node. From there, you will have a list of all genes in that strain. You can use the arrow buttons to find CTL0260.<br><br>6. <strong class="text-pink-light">I discovered that CTL0260 localizes to the pathogen-containing (inclusion) membrane. Now what?</strong><br>Navigate to the gene page for CTL0260 (See #5). Scroll down the page until you see the "Localizations" module. Under the Cellular Component section, click "Add Annotation". You will be guided through the annotation process at this point.<br><br>7. <strong class="text-pink-light">How do I authorize my account to use ChlamBase?</strong><br>Authorization is only required on gene specific pages. That means you will only see the authorization button appear if you are on the gene page describing 'CTL0260'. Once on that page, click the green button in the upper right corner. If you are a new user, make sure to click "Create Account" from the form. Otherwise, skip below to click authorize ChlamBase. Make sure you login with your MediaWiki details. Once you return to ChlamBase after logging in, notice that the login button will turn orange, and in the URL browser you see "authorized". You can now add annotations from ChlamBase!<br><br>8. <strong class="text-pink-light">It looks like I'm authorized but I still am not. What do I do now?</strong><br>Clear your cache. Then, navigate to <a href="https://www.wikidata.org" target="_blank">WikiData</a> and in the upper right corner, click "log in". Then go back to ChlamBase. If you are authorized, deauthorize. Then, reauthorize on ChlamBase and try to add your annotation again.<br><br>9. <strong class="text-pink-light">What is MediaWiki? What is WikiData?</strong><br>ChlamBase stores all of your annotations in an online database called WikiData. WikiData is an open, editable, user-curated knowledge graph of all kinds of information, not just Chlamydia! WikiData is built using MediaWiki (like wikipedia), and ChlamBase is the user interface that sits on top of the WikiData layer that allows easy access to read and write information for non coders.<br></div></div></div></div>